nominal/core/dataset_file.py

```diff
@@ -36,8 +50,69 @@ def _from_conjure(cls, dataset_file: scout_catalog.DatasetFile) -> Self:
             uploaded_at=upload_time,
             ingested_at=ingest_time,
             ingest_status=IngestStatus._from_conjure(dataset_file.ingest_status),
+            _clients=clients,
         )
 
+    def download(self, destination: pathlib.Path) -> pathlib.Path:
```

Comment:
IIUC destination is a directory, the return is the actual file path? let's maybe use `output_dir` or something so that it's clear it should be a dir?